src/cata_tiles.h

```diff
@@ -282,6 +282,7 @@ class cata_tiles
         int get_tile_width() const { return tile_width; }
         float get_tile_ratiox() const { return tile_ratiox; }
         float get_tile_ratioy() const { return tile_ratioy; }
+        tile_id_map get_tile_ids() const { return tile_ids; }
```

Comment:
And another even bigger performance hit is the copy of the `tile_ids`, which is done for **every single entry of the checked objects**. There are about 2800 items, 200 vehicle parts, 200 terrain and 100 furniture types and another bunch of monster types, you make a copy of this `tile_ids` for **every** single one of those things. Make this return a const reference and store the reference (as reference, not as copy) in the functions that use it instead of calling it in each iteration.

```C++
const tile_id_map & get_tile_ids() const { return tile_ids; }

auto const & tile_ids = tile_context->get_tile_ids();
```

Or even better, move the logic to check tile existence into `cata_tiles`:
```C++
bool cata_tiles::has_tile( std::string const & id ) const
{
    return tile_ids.count( id ) > 0;
}
```